src/Umbraco.Core/Compose/UnattendedInstallComponent.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Exceptions;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations.Install;
+
+namespace Umbraco.Core.Compose
+{
+    public class UnattendedInstallComponent : IComponent
+    {
+        private readonly IRuntimeState _runtimeState;
+        private readonly ILogger _logger;
+        private readonly DatabaseBuilder _databaseBuilder;
+
+        public UnattendedInstallComponent(IRuntimeState runtimeState, ILogger logger, DatabaseBuilder databaseBuilder)
+        {
+            _runtimeState = runtimeState ?? throw new ArgumentNullException(nameof(runtimeState));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _databaseBuilder = databaseBuilder ?? throw new ArgumentNullException(nameof(databaseBuilder));
+        }
+
+        public static event EventHandler<UnattendedInstallEventArgs> InstallCompleted;
```

Comment:
What is the reason for requiring a restart? Do we need it? If not this simplifies a lot of things and we don't need this event. We'll also have to translate all of this to netcore where we are going to remove all static events and where restarts don't really exist (only shutdowns which require manually restarting the web app).

Reply:
Since it's in an component it's running after the runtime level is determined, so we need to restart after the install is done to get into the `Run` level. We could do the install ealier in the boot phase and never enter the install level if unattended installs is configured.